homeassistant/components/philips_js/__init__.py

```diff
@@ -1 +1,131 @@
-"""The philips_js component."""
+"""The Philips TV integration."""
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Any, Callable, Dict, Optional
+
+from haphilipsjs import ConnectionFailure, PhilipsTV
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_VERSION, CONF_HOST
+from homeassistant.core import Context, HassJob, HomeAssistant, callback
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+PLATFORMS = ["media_player"]
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Philips TV component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Philips TV from a config entry."""
+
+    tvapi = PhilipsTV(entry.data[CONF_HOST], entry.data[CONF_API_VERSION])
+
+    coordinator = PhilipsTVDataUpdateCoordinator(hass, tvapi)
+
+    await coordinator.async_refresh()
```

Comment:
Should we raise `ConfigEntryNotReady` if the first update fails?

Reply:
But raising ConfigEntryNotReady won't be needed if the entry can start and be unavailable until next coordinator update, right?

Reply:
Nope. That would block the ability to turn the device on with external actions.

Reply:
Does it make sense to do if they don't have a turn on trigger?

Reply:
maybe not. but turn on trigger can be added later. nor does it hurt anything to add it. it works perfectly fine with checks if tv is available from time to time.